### DISCUSSION TITLE
Idea: Template group sets

### DISCUSSION THREAD

**Discussion by blue-dreamer** [ExpressionEngine/ExpressionEngine#2894] [Feature Requests]
Currently you can only use one "set" of template groups on any one site (I'm ignoring MSM here). What I'm proposing is the ability to have multiple sets of templates groups per site. One would be flagged as current, but a site (super) admin would be able to switch the current set at any time.

Here's some example uses:

**Normal and "light" template sets**
Popular sites may experience a sudden traffic surge and its "design" may be too heavy and slow down response times. Imagine if you could flick a switch to a "light" design version, e.g. stripping out weight so pages load faster.

**Multiple site designs**
You could have instances where different designs are needed for different audiences, perhaps you need variations for English and non English, per continent, or even night and day. Imagine if EE could automatically switch template sets based on something like IP address or customer preference. This could also extend for different languages where you need a different UI.

**Development**
Testing a new design for an existing site can be a pain because you often have to sync the database to get the latest data regularly. Imagine if you could add a new template set to the live site and test with live data in real time? In this case the set would be marked as super admin access only (and possibly certain client roles for previewing etc). When your new set is ready making it live it would be one click in the CP.

**Time based**
Need a different design for the winter holidays? Imagine if you could pre-schedule EE to automatically switching sets at certain times.

This does sound like the concept of changeable themes as other CMS's use, but with a lot more flexibility.

**Comment by robsonsobral** [comment]
Imagine if I could share a pack of templates across 69 MSM sites!

**Comment by blue-dreamer** [comment]
Yeah why not, some structure like this?

/set1
/set1/default-site
/set1/msm1
/set1/msm2
/set1/msm3
/set2
/set2/default-site
/set2/msm1
/set2/msm2
/set2/msm3
/set3
/set3/default-site
/set3/msm1
/set3/msm2
/set3/msm3

**Comment by robsonsobral** [comment]
Nopes, I was thinking on share the same ~~pack~~ set for several sites the same way we share field groups. We go to the design area and see more than one ~~pack~~ set available, which we could link to that site or not.

Instead of store the templates on `templates/site_shortname`, store them on ~~`templates/pack_name`~~ `template/set_name`. Too much?

**Comment by blue-dreamer** [comment]
@robsonsobral yes in principal that sounds like my idea, call it a "pack" or "set".

**Comment by robsonsobral** [comment]
Maybe, `theme` is a better name, given `system/ee/ExpressionEngine/Service/Theme`.

A first step would be to create a `theme_name` config override, which would change the directory name of the templates.

Even if we don't go further on this, I would love to have this config override available right now. May I create it, @TomJaeger ?

**Comment by robsonsobral** [comment]
(changing my mind about the name)
